lightly/api/api_workflow_upload_metadata.py

```diff
@@ -0,0 +1,196 @@
+            custom_metadata[_COCO_ANNOTATION_KEYS.images],
+            key=lambda x: x[_COCO_ANNOTATION_KEYS.images_filename]
+        )
+
+        # sort metadata by image id
+        custom_metadata[_COCO_ANNOTATION_KEYS.custom_metadata] = sorted(
+            custom_metadata[_COCO_ANNOTATION_KEYS.custom_metadata],
+            key=lambda x: x[_COCO_ANNOTATION_KEYS.custom_metadata_image_id]
+        )
+
+        # get a list of filenames for binary search
+        image_filenames = [
+            image[_COCO_ANNOTATION_KEYS.images_filename] for image in 
+            custom_metadata[_COCO_ANNOTATION_KEYS.images]        
+        ]
+
+        # get a list of image ids for binary search
+        metadata_image_ids = [
+            data[_COCO_ANNOTATION_KEYS.custom_metadata_image_id] for data in 
+            custom_metadata[_COCO_ANNOTATION_KEYS.custom_metadata]        
+        ]
+
+        # map filename to metadata in O(n * logn)
+        filename_to_metadata = {}
+        for filename in filenames:
+
+            image_index = bisect_left(image_filenames, filename)
+            if image_index == len(image_filenames):
+                raise RuntimeError(
+                    f'Image with filename {filename} does not exist in custom metadata!'
+                )
+
+            image = custom_metadata[_COCO_ANNOTATION_KEYS.images][image_index]
+            image_id = image[_COCO_ANNOTATION_KEYS.images_id]
+
+            metadata_index = bisect_left(metadata_image_ids, image_id)
+            if metadata_index == len(metadata_image_ids):
+                raise RuntimeError(
+                    f'Image with id {image_id} has no custom metadata!'
+                )
+
+            metadata = custom_metadata[_COCO_ANNOTATION_KEYS.custom_metadata][metadata_index]
+            filename_to_metadata[filename] = metadata
+
+        return filename_to_metadata
+
+
+    def upload_custom_metadata(self,
+                               custom_metadata: Dict,
+                               verbose: bool = False):
+        """Uploads custom metadata to the Lightly platform.
+
+        The custom metadata is expected in a format similar to the COCO annotations:
+        Under the key "images" there should be a list of dictionaries, each with
+        a file_name and id. Under the key "metadata" the custom metadata is stored
+        as a list of dictionaries, each with a image_id to match it to the image.
+
+        Example:
+            custom_metadata = {
+                "images": [
+                    {
+                        "file_name": "image0.jpg",
+                        "id": 0
+                    },
+                    {
+                        "file_name": "image1.jpg",
+                        "id": 1
+                    }
+                ],
+                "metadata": [
+                    {
+                        "image_id": 0,  // this metadata belongs to "image0.jpg"
+                        "custom_metadata_entry_0": 0.5,
```

Comment:
Same here. Remove comment. Add more meaningful examples, please. An ideal example reflects all the constraints. If we support integer + string + float we should have examples for all in here. If we support that different samples have different metadata I would also show this here.